yarn-project/end-to-end/src/fixtures/token_utils.ts

```diff
@@ -37,6 +37,17 @@ export async function expectTokenBalance(
   expectedBalance: bigint,
   logger: DebugLogger,
 ) {
+  // Check if the wallet
```

Comment:
if the wallet... ?

Reply:
Whoops, didn't save the file properly.

~~To address the overall comment, the `Circuit execution failed` error seems to happen when validating a read request in the kernel, so I assumed it's because we haven't synched the balances properly (kernel verification happens with a stale header). Even if PXE~~

You made me think, hope you're happy. The problem is in a misalignment between what's read in the simulation (that uses notes seen up to `aztecNode.getBlockNumber()`) and the kernels (that checks read requests up to `synchronizer.getSynchedBlockNumber`). I'm gonna test this theory by forcing simulation to use only the synched blockNumber!